core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -139,47 +140,51 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
 
   private def hour(millis: Long = System.currentTimeMillis()) = millis / 1000 / 60 / 60
 
-  def recoverIndex(): Future[Unit] = {
+  def recoverIndex(): Future[Long] = {
     if (downsampleConfig.enablePersistentIndexing) {
       partKeyIndex.getCurrentIndexState() match {
         case (IndexState.Empty, _)              =>
           logger.info("Found index state empty, bootstrapping downsample index")
-          recoverIndexInternal
+          recoverIndexInternal(None)
+        case (IndexState.Synced, checkpointMillis)             =>
+          logger.info("Found index state empty, bootstrapping downsample index")
+          recoverIndexInternal(checkpointMillis)
```

Comment:
probably it should be renamed to recoverIndexEmpty() which does not take millis, (it will figure out millis on its own)